client/core/core.go

```diff
@@ -5693,6 +5693,11 @@ func (c *Core) prepareTradeRequest(pw []byte, form *TradeForm) (*tradeRequest, e
 		return nil, codedError(walletErr, fmt.Errorf("FundOrder error for %s, funding quantity %d (%d lots): %w",
 			assetConfigs.fromAsset.Symbol, fundQty, lots, err))
 	}
+	defer func() {
+		if _, err := c.updateWalletBalance(fromWallet); err != nil {
```

Comment:
Verified the bug on master, and that it's not on 0.5 (checked both on mainnet).

There was a `c.updateAssetBalance(...)` loop in `(*Core).Trade`, which we seem to have lost with the refactoring for `TradeAsync`.

However, the following was previously making the updated asset ID list, and considering parent wallets because possible locked fee balance.

```go
	updated := assetMap{fromWallet.AssetID: struct{}{}}
	if isAccountRefund && fromWallet.parent != nil {
		updated[fromWallet.parent.AssetID] = struct{}{}
	}
	if isAccountRedemption && toWallet.parent != nil {
		updated[toWallet.parent.AssetID] = struct{}{}
	}
```

I'm seeing the *parent* parts of that logic are now in `sendTradeRequest`, but I think all the balance updates belong where you are putting them here in `prepareTradeRequest`.

Reply:
I let this slip by, but do we also need to consider any parent asset for fee reserves?

Reply:
You're probably right. I'll fix that.